File: Python_Solutions/vowels/vowels.py
# Write a function that returns
# the number of vowels in a String.

def vowel_count(string):
    count = 0
    vowels = ['a', 'e', 'i', 'o', 'u']
    for char in string.lower():
        if char in vowels:
            count += 1
    return count


# Alternate solution
# def vowel_count(string):
#     count = 0
#     charMap = {}
#     vowels = ['a', 'e', 'i', 'o', 'u']
#     for char in string.lower():
#         if char not in charMap:
#             charMap[char] = 1
#         else:
#             charMap[char] += 1
#     for vowel in vowels:
#         try:
#             count += charMap[vowel]
#         except:
#             pass
#     return count


print(vowel_count('Why do YOU Ask?'))

File: Js_Solutions/vowels/vowels.js
// Write a function that returns 
// the number of vowels in a String.

function vowelCount(str) {
    let count = 0;
    const vowels = ['a', 'e', 'i', 'o', 'u'];
    for (let char of str.toLowerCase()) {
        if (vowels.includes(char)) count++;
    }
    return count;
}


// Alternate solution
// function vowelCount(str) {
//     let charMap = {};
//     const vowels = ['a', 'e', 'i', 'o', 'u'];
//     let count = 0;
//     for (char of str.toLowerCase()) {
//         charMap[char] ? charMap[char]++ : charMap[char] = 1;
//     }
//     for (let i = 0; i < vowels.length; i++) {
//         if (charMap[vowels[i]]) {
//             count += charMap[vowels[i]]
//         }
//     }
//     return count;
// }

console.log(vowelCount('HI there!'));